lib/mayaUsd/fileio/shading/shadingModeUseRegistry.cpp

```diff
@@ -340,41 +347,64 @@ class UseRegistryShadingModeExporter : public UsdMayaShadingModeExporter
                 *mat = material;
             }
 
+            const TfToken& renderContext = context.GetExportArgs().renderContext;
+            SdfPath materialExportPath = materialPrim.GetPath();
+
             UsdShadeShader surfaceShaderSchema =
                 _ExportShadingDepGraph(
-                    material,
+                    materialExportPath,
                     context.GetSurfaceShaderPlug(),
                     context);
+            TfToken entryPointName = UsdShadeTokens->surface;
+
+            // For preview we will use the universal context, and for anything else we will prefix
+            if (renderContext != UsdMayaShadingModeTokens->preview) {
+                entryPointName = TfToken(
+                    TfStringPrintf("%s:%s", renderContext.GetText(), entryPointName.GetText())
+                        .c_str());
+            }
```

Comment:
We should **not** be doing this ourselves. This and maybe `CreateShaderOutputAndConnectMaterial()` should be refactored to use `CreateSurfaceOutput()`, `CreateDisplacementOutput()`, and `CreateVolumeOutput()` from `UsdShadeMaterial`, and we should pass the appropriate renderContext into that:
https://github.com/PixarAnimationStudios/USD/blob/5d1771d23b935740bcd5bfb985a92a6652e070c0/pxr/usd/usdShade/material.h#L324

And again, we're talking about the "universal" renderContext, so we should use that term rather than "preview" to avoid confusion.